src/MacroTools/UnitCategory.cs

```diff
@@ -10,6 +10,10 @@ public enum UnitCategory
     /// <summary>
     /// The unit can train boats.
     /// </summary>
-    Shipyard
+    Shipyard,
+    /// <summary>
+    /// This unit is a portal
+    /// </summary>
+    Portal
```

Comment:
Looking again at `UnitType`, I think the `Meta` property is exactly what we're looking for here; "If true, this unit should never be deleted." Can we use that instead of using `UnitCategory`?